ReactWindows/ReactNative.Net46/ReactPage.cs

```diff
@@ -15,21 +15,35 @@ namespace ReactNative
     /// </summary>
     public abstract class ReactPage : Page, IAsyncDisposable
     {
-        private readonly IReactInstanceManager _reactInstanceManager;
-
-        private bool _isShiftKeyDown;
-        private bool _isControlKeyDown;
+        private readonly Lazy<IReactInstanceManager> _reactInstanceManager;
```

Comment:
To ensure that the Dispatcher is available regardless of the order the object's methods are called we use a Lazy allocation on both the ReactInstance and RootView... This allows us to have a single Set for the dispatcher... It also allows us to instantiate the ReactPage in the app.xaml.cs as read-only without the need to initialize anything until it is needed...